src/app/components/ui/Icon/index.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import * as React from 'react';
+import UIIcon from '@material-ui/core/Icon';
+import AddIcon from '@material-ui/icons/Add';
+import PlaceIcon from '@material-ui/icons/Place';
+import { withStyles } from '@material-ui/core/styles';
+import theme from 'app/styles/material';
+import clsx from 'clsx';
+
+import 'app/styles/modules/iconography.css';
+
+/* Types
+type Props = {
+  $white: true | false,
+  $button: true | false,
+  fontSize: "inherit" | "default" | "small" | "large" | "tiny",
+  color: "inherit" | "primary" | "secondary" | "action" | "error" | "disabled",
+  icon: "acciones"
+  | "aire" // AAC
+  | "alerta"
+  | "arrow-drop-down"
+  | "arrow-drop-up"
+  | "arrow_left"
+  | "arrow_less"
+  | "arrow_more"
+  | "arrow_right"
+  | "atras"
+  | "borrar"
+  | "caldera" //JUN
+  | "calendario"
+  | "camara"
+  | "cerrar"
+  | "cif"
+  | "clip"
+  | "cuadre-de-caja"
+  | "defectos"
+  | "direccion"
+  | "dto"
+  | "editr"
+  | "estado-aviso"
+  | "factura"
+  | "filtro"
+  | "firma"
+  | "historial"
+  | "info"
+  | "intervencion"
+  | "listado"
+  | "lupa"
+  | "mail"
+  | "mantenimiento"
+  | "menu"
+  | "movil"
+  | "observaciones"
+  | "papelera"
+  | "perfil"
+  | "phone"
+  | "piezas"
+  | "salir"
+  | "servicios"
+  | "sort"
+  | "toggle-off"
+  | "toggle-on"
+  | "usuario"
+  | "ver"
+  | "add"
+};*/
+
+const styles = {
+  root:{
+    width: 'auto',
+    height: 'auto',
+    paddingTop: '1px',
+    lineHeight: '1',
+    color: theme.palette.common.white
+  },
+  fab:{
+    position: 'relative',
+    top:'-9px'
+  }
+};
+
+
+const Icon = ( {
+  color,
+  classes,
+  $button,
+  fontSize,
+  icon,
+  fab=false,
+  button=false,
+  ...props
+} ) => {
+  icon = (icon === "JUN" && 'caldera') || icon;
+  icon = (icon === "AAC" && 'aire') || icon;
+  const classNames= clsx(`icon-${icon}`,fab && classes.fab);
+  return(
+    <React.Fragment>
+    {
+      !icon
```

Comment:
In general, its not a good idea to nest ternary operators, its hard to read. Better something like
```suggestion
      if (!icon) return (
         <AddIcon color={color} />
       );
       
       if (icon === 'place') return (
          <PlaceIcon color={color} />
       );
       
       return (
           <UIIcon
              color={ color }
              classes={$button && classes}
              fontSize={ fontSize === "tiny" ? "inherit" : fontSize }
              {...props}
            >
              <span className={classNames}></span>
          </UIIcon>
       );
```